From eec0a9685f270e3df08cc8017e476b2a5a83e985 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 14 Nov 2017 13:18:46 +0800
Subject: [PATCH 2803/5242] MLK-16813 drm/imx: core: Set driver data to NULL
 when we bailout from ->bind()

commit  e7fb6019c50f06ef6913857e820b6d552f13975a from
https://source.codeaurora.org/external/imx/linux-imx.git

We need to set driver data to NULL when we bailout from ->bind(),
otherwise it would be leaked to the system power management operations
and cause invalid driver data being used there.

Reported-by: Anson Huang <Anson.Huang@nxp.com>
Fixes: 54db5decce17 ("drm/imx: drop deprecated load/unload drm_driver ops")
Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/imx-drm-core.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/imx/imx-drm-core.c b/drivers/gpu/drm/imx/imx-drm-core.c
index 030d251..28dfa63 100644
--- a/drivers/gpu/drm/imx/imx-drm-core.c
+++ b/drivers/gpu/drm/imx/imx-drm-core.c
@@ -305,6 +305,7 @@ static int imx_drm_bind(struct device *dev)
 #endif
 	component_unbind_all(drm->dev, drm);
 err_kms:
+	dev_set_drvdata(dev, NULL);
 	drm_mode_config_cleanup(drm);
 err_unref:
 	drm_dev_unref(drm);
-- 
1.7.9.5

